backend/src/v5/processors/users.js

```diff
@@ -98,9 +103,18 @@ Users.deleteApiKey = deleteApiKey;
 
 Users.getUserByUsername = getUserByUsername;
 
-Users.getAvatar = getAvatar;
+Users.getAvatarStream = (username) => getFileAsStream('admin', AVATARS_COL_NAME, username);
+
+Users.uploadAvatar = async (username, avatarBuffer) => {
+	const existingRef = await getRefEntry('admin', AVATARS_COL_NAME, username);
+	if (existingRef) {
+		await removeRef('admin', AVATARS_COL_NAME, username);
+		await removeFiles('admin', AVATARS_COL_NAME, existingRef.type, [existingRef.link]);
+	}
 
-Users.uploadAvatar = uploadAvatar;
+	const refInfo = await storeFile('admin', AVATARS_COL_NAME, avatarBuffer);
```

Comment:
I think it's better of storeFile does all of the work, i.e. let File manager check if a ref already exists for that filename and update appropriately.

Basically this whole function should just be calling fileManager with the right parameters.

The whole point of fileManager is to deal with all the inner works of storing the file (i.e. talk to external file handlers, creating and managing file refs) so everyone else doesn't need to.